packages/amplify-category-geo/src/index.ts

```diff
@@ -32,9 +32,19 @@ export const executeAmplifyCommand = async (context: $TSContext) => {
     }
 };
 
-export const handleAmplifyEvent = (context: $TSContext, args: $TSObject) => {
-    printer.info(`${category} handleAmplifyEvent to be implemented`);
-    printer.info(`Received event args ${args}`);
+export const handleAmplifyEvent = async (context: $TSContext, args: $TSAny) => {
+  switch (args.event) {
+    case 'PrePush':
+      if((await checkAnyGeoResourceExists()) && !verifySupportedRegion()) {
+        const errMessage = 'Failed to create Geo resources';
+        await context.usageData.emitError(new Error(errMessage));
+        printer.info('Remove Geo resources using "amplify remove geo" and retry "amplify push"');
+        exitOnNextTick(1);
```

Comment:
if we throw an error here will it stop the push? If so, that would be a better way to control the flow rather than exit on next tick